python/taichi/lang/ast/ast_transformer.py

```diff
@@ -1157,6 +1174,22 @@ def build_static_for(ctx, node, is_grouped):
         else:
             build_stmt(ctx, node.iter)
             targets = ASTTransformer.get_for_loop_targets(node)
+            iter_time = 0
+            if ti_unroll_limit:
```

Comment:
Can we move this inside the `for` on line 1193?

Reply:
sorry, forget commit `break` when exceed limit; and I sugguest not nest this check inside real complie block, because if loop is huge, program must check this limit in every loop, and inject this check inside is not look good, I think histing this behavior outside is ok.

Reply:
> Can we move this inside the `for` on line 1193?

sorry, forget commit `break` when exceed limit; and I sugguest not nest this check inside real complie block, because if loop is huge, program must check this limit in every loop, and inject this check inside is not look good, I think histing this behavior outside is ok.